File: backend/app/schemas/certmatrix.py
"""資格マトリクススキーマ"""
from pydantic import BaseModel
from app.schemas.employee import EmployeeListItem


class CertInfo(BaseModel):
    id: str
    name: str


class EngineerCertEntry(BaseModel):
    cert_id: str
    obtained_at: str | None = None
    expires_at: str | None = None
    expiry_status: str | None = None  # SOON / VALID / NO_EXPIRY


class CertMatrixCategory(BaseModel):
    category: str          # LANGUAGE / CLOUD / PM / NETWORK / SECURITY / OTHER
    certifications: list[CertInfo]


class CertEngineerRow(BaseModel):
    employee: EmployeeListItem
    certs: dict[str, EngineerCertEntry]   # cert_master_id -> entry


class CertMatrixResponse(BaseModel):
    categories: list[CertMatrixCategory]
    engineers: list[CertEngineerRow]

File: frontend/src/api/certmatrix.ts
/**
 * 資格マトリクス API クライアント
 */
import apiClient from './client'
import type { EmployeeListItem } from './employees'

export interface CertInfo {
  id: string
  name: string
}

export interface CertMatrixCategory {
  category: string  // LANGUAGE | CLOUD | PM | NETWORK | SECURITY | OTHER
  certifications: CertInfo[]
}

export interface EngineerCertEntry {
  cert_id: string
  obtained_at: string | null
  expires_at: string | null
  expiry_status: string | null  // SOON | VALID | NO_EXPIRY
}

export interface CertEngineerRow {
  employee: EmployeeListItem
  certs: Record<string, EngineerCertEntry>  // cert_master_id -> entry
}

export interface CertMatrixResponse {
  categories: CertMatrixCategory[]
  engineers: CertEngineerRow[]
}

export interface CertMatrixParams {
  location?: string
  category?: string
  free_only?: boolean
  search?: string
}

export const certMatrixApi = {
  get: (params?: CertMatrixParams) =>
    apiClient.get<CertMatrixResponse>('/certifications/matrix', { params }).then(r => r.data),
}
